operator/pkg/reconciliation/check_nodes.go

```diff
@@ -0,0 +1,183 @@
+// Copyright DataStax, Inc.
+// Please see the included license file for details.
+
+package reconciliation
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (rc *ReconciliationContext) GetPVCForPod(podNamespace string, podName string) (*corev1.PersistentVolumeClaim, error) {
+	pvcFullName := fmt.Sprintf("%s-%s", PvcName, podName)
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := rc.Client.Get(rc.Ctx, types.NamespacedName{Namespace: podNamespace, Name: pvcFullName}, pvc)
+	if err != nil {
+		rc.ReqLogger.Error(err, "error retrieving PersistentVolumeClaim")
+		return nil, err
+	}
+
+	return pvc, nil
+}
+
+func (rc *ReconciliationContext) DeletePvcIgnoreFinalizers(podNamespace string, podName string) (*corev1.PersistentVolumeClaim, error) {
```

Comment:
Ugh. What I want to say we should do here is, if there is a non-blocking way to delete the PVC, just do that and then delete the Pod. When the terminates, the finalizer `kubernetes.io/pvc-protection` should be removed automagically and deletion of the PVC should then complete. This is making me wonder if we should _just_ delete the pod when we see a node taint, and then handle Pods that cannot be scheduled due to the node affinity of their PVC separately (i.e. don't do a replace, just delete the pod, if we see a stuck pod due to a PVC issue, then delete the PVC and do a replace).  We may end up just leaving it this way for now though...

Reply:
Err... also, deleting the PVC out from under the pod might cause the pod to be unable to terminate cleanly... not sure if that matters in the case of Cassandra...